### ISSUE TITLE
Dynamic Ref fails to update

### ISSUE THREAD

**Issue by kanakiyajay** [Automattic/mongoose#4104]
**Environment**

```
Windows 7 64 bit
node v4.2.4
npm v2.14.12
mongoose v4.4.14
mongodb v3.2.14
```

**Code:**

```js
'use strict';

var mongoose = require('mongoose'),
    Schema = mongoose.Schema,
    should = require('should');

// Connect to database
mongoose.connect('mongodb://localhost/tp-dev');

var PersonSchema = new Schema({
  name: { type: String, default: 'Jay'}
});

var AnimalSchema = new Schema({
  name: { type: String, default: 'dog'}
});

var ThingSchema = new Schema({
  created_by_model: { type: String },
  created_by: { type: mongoose.Schema.Types.ObjectId, refPath: 'created_by_model'},
});

var Thing = mongoose.model('Thing', ThingSchema);
var Person = mongoose.model('Person', PersonSchema);
var Animal = mongoose.model('Animal', AnimalSchema);

Person.create({}, function(err, person) {
  console.log(person.name); // 'Jay'
  Animal.create({}, function(err, animal) {
    console.log(animal.name); // 'dog'
    Thing.create({
      created_by_model: 'Person',
      created_by: person._id
    }, {
      created_by_model: 'Animal',
      created_by: animal._id
    }, function(err, t1, t2) {
      Thing
        .find({})
        .populate('created_by')
        .exec(function(err, things) {
          
          console.log(things);
          /*
          [ { __v: 0,
            created_by: 57220b022385878c33ca8e8a, // This is not updated
            created_by_model: 'Person',
            _id: 57220b022385878c33ca8e8c },
          { __v: 0,
            created_by: { name: 'dog', __v: 0, _id: 57220b022385878c33ca8e8b },
            created_by_model: 'Animal',
            _id: 57220b022385878c33ca8e8d } ]
            */

        });
    });
  });
});
```

**TL;DR**
The Things created_by fails to update in above case.
This is done according to the documentation, kindly close the issue if I have implemented it incorrectly.
